pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -25,10 +26,51 @@
 from .optimization_utils import find_next_gate
 
 
+def _ops_equal(op1, op2):
+    """Checks if two operators are equal up to class, data, hyperparameters, and wires"""
+    return (
+        op1.__class__ is op2.__class__
+        and (op1.data == op2.data)
+        and (op1.hyperparameters == op2.hyperparameters)
+        and (op1.wires == op2.wires)
+    )
```

Comment:
We should just add this to the operator class ;)

Reply:
There's a good-first-issue open for a `qml.equal` comparison function.

Reply:
I always wonder if it is super trivial or super hard. The hyperparams dict could in principle contain anything. 

Maybe one should implement such a method so that it _tries_ to compare hyperparams, but errors if something is not comparable.